specification/cognitiveservices/data-plane/Language/preview/2021-11-01-preview/analyzeconversations.json

```diff
@@ -0,0 +1,560 @@
+            "$ref": "./examples/conversations/Conversation_project.json"
+          },
+          "Orchestrator direct target result": {
+            "$ref": "./examples/conversations/Orchestrator_direct_target.json"
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "ConversationAnalysisOptions": {
+      "type": "object",
+      "description": "The request body.",
+      "required": [
+        "query"
+      ],
+      "properties": {
+        "query": {
+          "description": "The conversation utterance to be analyzed.",
+          "type": "string"
+        },
+        "directTarget": {
+          "type": "string",
+          "description": "The name of the target project this request is sending to directly."
+        },
+        "language": {
+          "type": "string",
+          "description": "The language to use in this request. This will be the language setting when communicating with all other target projects."
+        },
+        "verbose": {
+          "type": "boolean",
+          "description": "If true, the service will return more detailed information in the response."
+        },
+        "isLoggingEnabled": {
+          "description": "If true, the query will be kept by the service for customers to further review, to improve the model quality.",
+          "type": "boolean"
+        },
+        "parameters": {
+          "type": "object",
+          "description": "A dictionary representing the input for each target project.",
+          "additionalProperties": {
+            "$ref": "#/definitions/AnalysisParameters"
+          }
+        }
+      }
+    },
+    "AnalysisParameters": {
+      "type": "object",
+      "description": "This is the parameter set of either the Orchestration project itself or one of the target services.",
+      "required": [
+        "targetKind"
+      ],
+      "discriminator": "targetKind",
+      "properties": {
+        "targetKind": {
+          "type": "string",
+          "description": "The type of a target service.",
+          "enum": [
+            "luis",
+            "conversation",
+            "question_answering",
+            "non_linked"
+          ],
+          "x-ms-enum": {
+            "name": "targetKind",
+            "modelAsString": true
+          }
+        },
+        "apiVersion": {
+          "type": "string",
+          "description": "The API version to use when call a specific target service."
+        }
+      }
+    },
+    "NoneLinkedTargetIntentResult": {
+      "type": "object",
+      "description": "A wrap up of non-linked intent response.",
+      "x-ms-discriminator-value": "non_linked",
+      "allOf": [
+        {
+          "$ref": "#/definitions/TargetIntentResult"
+        }
+      ],
+      "properties": {
+        "result": {
+          "type": "object",
+          "$ref": "#/definitions/ConversationResult",
+          "description": "The actual response from a Conversation project.",
+          "x-nullable": true
```

Comment:
You should avoid the use of `x-nullable: true`. Properties with no value should simply be omitted from the payload.

Reply:
I want to clarify this, though. While I agree with @mikekistler, I know from experience that quite a few implements don't or can't (like some using common OData libraries), in which case you should keep `"x-nullable": true` because, if you don't, code generation can be wrong.

Reply:
If the service is using a library that is unable to omit the value, we should have a conversation on how to address/fix that. A nullable optional property is very unfortunate to have in the Service API.

Reply:
I read .NET? :-)

If service lib does return `"property": null`, and .NET does require `x-nullable` to take a `null` property, we seems not able to remove it?

Reply:
More problematic for the default tag is that it won't work with the current cognitiveservices/Language RP. Despite being in a single RP meant for a Language pillar unification including - originally, but no longer - release schedules, each swagger (at least by the first dash-delimited file name term) is actually a distinct SDK. They will ship on different schedules, though we want to try to maintain fewer api-versions across them. So either we need to rethink how we tag (default or otherwise) and validation such swaggers, or we need to separate them into separate RPs - distinct from the old ones, since these are actually new and contain potentially many breaking changes.

Reply:
Agree, Heath. Josehpine Qiu is also reviewing the readme structure, though her main attention is for MSDocs. SDK with Renhe & Laurent is also reviewing the same thing for data-plane SDK purpose.

Current validation structure mostly works fine for management (though still corner case for resources and compute), but it very likely not working good for data-plane.

Reply:
Actually in the service's response, there is no null values will be returned. We ignore all null values in the response.

Weidong mentioned this: ”I've added SignedOff label, as this is more on SDK than API.“, and this is the actual case.

Can we make a decision on this right now? We will take the decision in our next version.

Reply:
If service does not return `null`, there seems to be no reason to have these `x-nullable`? (I seems seeing this being fixed in https://github.com/Azure/autorest.java/pull/1377 on Mike's recommendation but that PR get closed)

As we already have 2 approvals, we can merge when requested (probably had to bypass Avocado and LintDiff).